drivers/serial/uart_sifive.c

```diff
@@ -334,6 +339,13 @@ static int uart_sifive_init(const struct device *dev)
 	cfg->cfg_func();
 #endif
 
+#ifdef CONFIG_PINCTRL
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+#endif
```

Comment:
does driver work at all without pinctrl? why is this made optional (same applies to others)?

Reply:
This change was necessary for the `hifive_unleashed` and `hifive_unmatched` boards, as they weren't using the old pinmux driver before.

Reply:
They don't do it at all, if I'm correct. I've re-checked this with their SoCs manuals (FU540 and FU740 respectively), and the section that would describe IOFs is empty.